src/Event/AuthorExtractDataEvent.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content\Event;
+
+use Drupal\Core\Cache\CacheableResponseTrait;
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Link;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event used to extract sub-entity type Links individually for each bundle.
+ */
+class AuthorExtractDataEvent extends Event {
+
+  use CacheableResponseTrait;
+
+  /**
+   * The name of the event.
+   */
+  const EXTRACT_AUTHOR_LINKS = 'oe_content.event.get_author_entity_links';
```

Comment:
```suggestion
  const EXTRACT_AUTHOR_LINKS = 'oe_content.get_author_entity_links';
```
Event definition should be in a separate class.